src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1304,6 +1305,44 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(self, plan: SnowflakePlan):
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        if isinstance(plan, SelectTableFunction) and isinstance(
+            plan.snowflake_plan.source_plan, TableFunctionJoin
+        ):
+            if (
+                plan.snowflake_plan.source_plan.right_cols == ["*"]
+                and len(plan.snowflake_plan.children_plan_nodes) == 1
+            ):
+                child_plan = plan.snowflake_plan.children_plan_nodes[0]
+                if isinstance(child_plan, Selectable):
+                    child_plan = child_plan.snowflake_plan
+                plan.snowflake_plan.source_plan.right_cols = (
+                    plan.snowflake_plan.quoted_identifiers[
+                        len(child_plan.quoted_identifiers) :
+                    ]
+                )
+                new_plan = self.session._analyzer.resolve(
+                    plan.snowflake_plan.source_plan
+                )
+                plan._snowflake_plan = new_plan
+                return plan
+        plan_2_resolve = None
+        for node in plan.children_plan_nodes:
+            plan_2_resolve = (
+                self.find_table_function_in_sql_tree(node) or plan_2_resolve  # type: ignore
+            )
```

Comment:
Try using BFS. In the past, we have discovered that using DFS for snowpark plans can generate `MaxRecursionDepthExceeded` errors